x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -134,6 +135,8 @@ public LogicalPlan optimize(LogicalPlan verified) {
                 // needs to occur before BinaryComparison combinations (see class)
                 new PropagateEquals(),
                 new CombineBinaryComparisons(),
+                // Geo
```

Comment:
There's already a rule that does this for `BinaryOperator`s - if `StDistance` extends that calls (and it looks like it can), the swapping will happen automatically.